.github/workflows/build-deps.yml

```diff
@@ -3,20 +3,230 @@ name: Build emsdk dependencies
+
+    runs-on: [self-hosted, Windows, ARM64, WASM]
+
+    steps:
+    - name: Build Binaryen for Arm64
+      shell: cmd
+      run: |
+        git clone https://github.com/WebAssembly/binaryen.git --single-branch --depth 1
+        cd binaryen
+        git submodule init
+        git submodule update
+        if exist bin rmdir /s /q bin
+        call "%ProgramFiles%\Microsoft Visual Studio\2022\Preview\VC\Auxiliary\Build\vcvarsall.bat" arm64
+        cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER_LAUNCHER='ccache' .
+        ninja
+
+
+    - name: Archive binaryen-arm64.zip
+      uses: actions/upload-artifact@v3
+      with:
+        name: binaryen-arm64.zip
+        path: binaryen/binaryen-arm64.zip
+        retention-days: 1
+
+  build-nodejs:
+    if: ${{ inputs.nodejs_build }}
+
+    runs-on: [self-hosted, Windows, ARM64, WASM]
+
+    steps:
+
+    - name: Build Node.js arm64
+      run: |
+        Set-PSDebug -Trace 1
+        git clone https://github.com/Windows-on-ARM-Experiments/node.git --single-branch -b fix-arm64-compilation --depth 1
+        cd node
+        .\vcbuild openssl-no-asm arm64
+
+  build-emscripten:
+    if: ${{ inputs.emscripten_build || inputs.wasm_binaries_build }}
+
+    runs-on: [ubuntu-latest]
+
+    steps:
+
+    - name: Build Emscripten
+      run: |
+        git clone https://github.com/emscripten-core/emscripten.git --single-branch --depth 1
+        cd emscripten
+        ./tools/install.py ../emscripten_package
+        npm ci --production --no-optional ../emscripten_package
+        cd ../emscripten_package
+        zip -r ../emscripten_package.zip .
+
+    - name: Archive emscripten_package.zip
+      uses: actions/upload-artifact@v3
+      with:
+        name: emscripten_package.zip
+        path: emscripten_package.zip
+        retention-days: 1
+
+  build-wasm-binaries:
+    if: ${{ inputs.wasm_binaries_build }}
+    needs: [build-emscripten, build-binaryen, build-llvm-webassembly-arm64]
+
+    runs-on: [windows-latest]
+
+    steps:
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: emscripten_package.zip
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: llvm-arm64.zip
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: binaryen-arm64.zip
+
+    - name: Pack wasm-binaries for Arm64
       run: |
-        set
         ls
+        New-Item -ItemType Directory -Force -Path emsdk/emscripten
+        Expand-Archive -LiteralPath emscripten_package.zip -DestinationPath emsdk\emscripten
+        Expand-Archive -LiteralPath llvm-arm64.zip -DestinationPath emsdk
+        Expand-Archive -LiteralPath binaryen-arm64.zip -DestinationPath emsdk
+        cd emsdk
+        Compress-Archive -Path * -DestinationPath wasm-binaries-arm64
+
+    - name: Archive wasm-binaries-arm64.zip
+      uses: actions/upload-artifact@v3
+      with:
+        name: wasm-binaries-arm64.zip
+        path: emsdk/wasm-binaries-arm64.zip
```

Comment:
When I've compared the structure of the current stable x64 wasm-binaries.zip (e.g, https://storage.googleapis.com/webassembly/emscripten-releases-builds/win/a8c3b314d61e2bb98581d522f858132b2fc21488/wasm-binaries.zip) with the latest result of the pipeline (https://github.com/Windows-on-ARM-Experiments/emsdk/suites/10865410545/artifacts/547955478) there are the following differences:

1.  Double zipped archive (see the previous comment).
2.  The upstream archive starts with an `install` root folder.
3.  The upstream contains `emscripten_config_upstream` Python script with some additional install steps.
4.  Our package contains additional and maybe unnecessary? `.manfest` files.
5.  Our package contains some additional clang linker related binaries - this could explain our issues with linking the samples.
6.  The upstream package contains CLang headers in `lib/clang/15.0.0/include` while our package in `clang/17/include`. I've seen some unit test compilation errors that might be related to that.
7.   `escripten` directory in our package is missing `node_modules` directory with lot of Node packages.